### PR TITLE
feat(sql): vectorized aggregation of `short` columns

### PR DESCRIPTION
**Pull request by bluestreak01**
Vector accelration of `min()`, `max()`, `sum()` and `avg()` functions

### CODE REVIEW

File: core/src/main/c/share/vec_int_key_agg.cpp [hunk 1]
```diff
@@ -286,7 +289,62 @@ static jboolean kIntMinLong(to_int_fn to_int, jlong pRosti, jlong pKeys, jlong p
...
+        if (PREDICT_FALSE(res.second)) {
+            if (PREDICT_FALSE(res.first == UL_MAX)) {
+                return JNI_FALSE;
+            }
+            *reinterpret_cast<int32_t *>(pKey) = key;
+            *reinterpret_cast<jlong *>(pVal) = val;
+        } else {
+            const jlong old = *reinterpret_cast<jlong *>(pVal);
+            if (old != L_MIN) {
+                *reinterpret_cast<jlong *>(pVal) = MIN(val, (jshort) old);
```
Comment (ideoma) [inline core/src/main/c/share/vec_int_key_agg.cpp:312]: I think it should be
```suggestion
             if (val != I_MIN) {
                 *reinterpret_cast<jlong *>(pVal) = MIN(val, (jshort) old);
             }
```

File: core/src/main/c/share/vec_agg.h [hunk 2]
```diff
@@ -243,6 +243,41 @@ JNIEXPORT jlong JNICALL Java_io_questdb_std_Vect_ ## func(JNIEnv *env, jclass cl
 \
 }
 
+typedef int64_t LongShortVecFuncType(int16_t *, int64_t);
```
[the remark this answers is not in the document]
Comment (bluestreak01) [reply core/src/main/c/share/vec_agg.h:246]: will delete in separate PR

**Comment by ideoma** [comment]
LGTM

**Review by ideoma — APPROVED** [review]

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 148 / 160 (92.50%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/griffin/engine/groupby/vect/MinShortVectorAggregateFunction.java|31|36|86.11%|
|:large_blue_circle:|io/questdb/griffin/engine/groupby/vect/AvgShortVectorAggregateFunction.java|45|48|93.75%|
|:large_blue_circle:|io/questdb/griffin/engine/groupby/vect/MaxShortVectorAggregateFunction.java|29|31|93.55%|
|:large_blue_circle:|io/questdb/griffin/engine/groupby/vect/SumShortVectorAggregateFunction.java|34|36|94.44%|
|:large_blue_circle:|io/questdb/PropServerConfiguration.java|2|2|100.00%|
|:large_blue_circle:|io/questdb/network/QueryPausedException.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/cairo/DatabaseSnapshotAgentImpl.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/cairo/sql/NetworkSqlExecutionCircuitBreaker.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/griffin/SqlCodeGenerator.java|4|4|100.00%|

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 148 / 160 (92.50%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/griffin/engine/groupby/vect/MinShortVectorAggregateFunction.java|31|36|86.11%|
|:large_blue_circle:|io/questdb/griffin/engine/groupby/vect/AvgShortVectorAggregateFunction.java|45|48|93.75%|
|:large_blue_circle:|io/questdb/griffin/engine/groupby/vect/MaxShortVectorAggregateFunction.java|29|31|93.55%|
|:large_blue_circle:|io/questdb/griffin/engine/groupby/vect/SumShortVectorAggregateFunction.java|34|36|94.44%|
|:large_blue_circle:|io/questdb/PropServerConfiguration.java|2|2|100.00%|
|:large_blue_circle:|io/questdb/network/QueryPausedException.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/cairo/DatabaseSnapshotAgentImpl.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/cairo/sql/NetworkSqlExecutionCircuitBreaker.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/griffin/SqlCodeGenerator.java|4|4|100.00%|

**Pull request merged**